sympy/polys/rationaltools.py

```diff
@@ -83,3 +83,68 @@ def _together(expr):
         return expr
 
     return _together(sympify(expr))
+
+
+def thiele_interpolate(u, v, var=symbols('x'), simplify=True):
+    """
+    Build a rational function from a finite set of inputs in vector u
+    and their function values in vector v (both vectors must be of equal
+    lengths).
+
+    This algorithm might encounter division by zero
+    when values are equally spaced. For such cases, a better algorithm is
+    sympy.polys.polyfuncs.rational_interpolate.
```

Comment:
Is it always true that the `rational_interpolate(dat, n) == thiele_interpolate(*dat)` if n is the degree of the Thiele numerator?